source/Cosmos.System2/FileSystem/CosmosVFS.cs

```diff
@@ -944,9 +944,25 @@ public override void SetFileSystemLabel(string aDriveId, string aLabel)
         /// <exception cref="NotSupportedException">Thrown when FAT type is unknown.</exception>
         public override void Format(string aDriveId, string aDriveFormat, bool aQuick)
         {
-            var xFs = GetFileSystemFromPath(aDriveId);
+            if (aDriveId.Length != 1 && !char.IsDigit(aDriveId[0]))
+            {
+                throw new ArgumentException("aDriveId is not a drive number.");
+            }
+
+            try
+            {
+                var xFs = GetFileSystemFromPath(aDriveId);
 
-            xFs.Format(aDriveFormat, aQuick);
+                xFs.Format(aDriveFormat, aQuick);
+            }
+            catch
+            {
+                var id = int.Parse(aDriveId);
+                var partition = mPartitions[id];
+                var xSize = (long)(partition.BlockCount * partition.BlockSize / 1024 / 1024);
+                var fs = FatFileSystem.CreateFatFileSystem(partition, id + ":\\", xSize);
```

Comment:
Why arnt you checking `aDriveFormat` for the correct value? While currently there are no other possible values, this prevents us from having breaking changes in the future.